### PR TITLE
Aero workflow components

### PR DESCRIPTION
**Pull request by feyzullah**
### Actions
* Create Appointment
* Create Company
* Create Contact
* Create Email Aero
* Create Task
* Create Vault Entry
* Find Company

### Sources
* New Company

### CODE REVIEW

File: components/aero_workflow/sources/new-company/new-company.mjs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import common from "../common/common.mjs";
+
+export default {
+  ...common,
+  key: "aero_workflow-new-company",
```
Comment (andrewjschuang) [inline components/aero_workflow/sources/new-company/new-company.mjs:5]: Can you please refer to this docs on [component key patterns](https://pipedream.com/docs/components/guidelines/#component-key-pattern)?

File: components/aero_workflow/sources/new-company/new-company.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        resourceFn: this.app.getCompanies,
+        resourceKey: "companies",
+      };
+    },
+    getComparable(item) {
+      return item.id;
+    },
+    getMeta(item) {
+      return {
+        id: item?.id,
```
Comment (andrewjschuang) [inline components/aero_workflow/sources/new-company/new-company.mjs:24]:
```suggestion
        id: item.id,
```

File: components/aero_workflow/common/utils.mjs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  async asyncPropHandler({
+    resourceFn, labelVal, params, resourceKey,
+  } = {}) {
+    let resp;
+    resp = await resourceFn({
+      params: {
+        ...params,
+      },
+    });
```
Comment (andrewjschuang) [inline components/aero_workflow/common/utils.mjs:31]:
```suggestion
    const resp = await resourceFn({
      params,
    });
```
Minor

**Review by andrewjschuang — COMMENTED** [review]
Thanks! I've left a few comments below.

Sorry, I can't quite grasp what's happening with the split(".") and reduce parts. Can you please explain to me?

**Comment by feyzullah** [comment]
> Thanks! I've left a few comments below.
> 
> Sorry, I can't quite grasp what's happening with the `split(".")` and `reduce` parts. Can you please explain to me?
Hi @andrewjschuang, actually it is done to extract resources if they are nested. For example if the resources is fetched like `{ response: { innerResponse: [ ...items ] } }` it extracts items with a key like `resources.innerResources`. There were no inner keys for these components but I kept it. If you would like me to remove them I will. Thank you very much

**Comment by andrewjschuang** [comment]
> Hi @andrewjschuang, actually it is done to extract resources if they are nested. For example if the resources are fetched like { response: { innerResponse: [ ...items ] } } it extracts items with a key like resources.innerResources. There were no inner keys for these components but I kept it. If you would like me to remove them I will. Thank you very much

Ah ok, I think next time if it's not necessary for a specific app/component it is better to make it as simple as possible. But since you've already put it here, then it's ok to leave it.

**Review by andrewjschuang — APPROVED** [review]
Thanks, LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Aero_Workflow_1753.pdf](https://github.com/PipedreamHQ/pipedream/files/11178308/Aero_Workflow_1753.pdf)

**Pull request merged**